core/src/main/scala/kafka/server/mirror/MirrorFetcherThread.scala

```diff
@@ -68,18 +68,6 @@ class MirrorFetcherThread(name: String,
                                 mirrorName) {
   this.logIdent = logPrefix
 
-  // Visible for testing
-  private[mirror] val partitionsWithNewHighWatermark = mutable.Buffer[TopicPartition]()
-
-  override def doWork(): Unit = {
-    super.doWork()
-    // Complete delayed fetch requests for consumers waiting on mirrored partitions
-    if (partitionsWithNewHighWatermark.nonEmpty) {
-      replicaMgr.completeDelayedFetchRequests(partitionsWithNewHighWatermark.toSeq)
```

Comment:
Forgot to say, because we don't have delayedFetch in mirror fetch, we can safely remove them here.